File: app/src/main/java/com/dicoding/projectcapstone/kategori/KategoriActivity.java
package com.dicoding.projectcapstone.kategori;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.dicoding.projectcapstone.R;
import java.util.ArrayList;

public class KategoriActivity extends AppCompatActivity {

    private RecyclerView foodList;
    private RecyclerView drinkList;
    private ArrayList<Kategori> kategoriList;
    private KategoriAdapter kategoriAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_kategori);

        // Inisialisasi RecyclerView
        foodList = findViewById(R.id.food_list);
        drinkList = findViewById(R.id.drink_list);

        // Membuat data kategori
        kategoriList = new ArrayList<>();
        kategoriList.add(new Kategori(R.drawable.ic_food, "Food", 4.5, "2 km", "$10", true));
        kategoriList.add(new Kategori(R.drawable.ic_drink, "Drink", 4.0, "1.5 km", "$5", false));


        // Inisialisasi Adapter
        kategoriAdapter = new KategoriAdapter(kategoriList);

        // Atur RecyclerView untuk daftar makanan
        foodList.setAdapter(kategoriAdapter);
        foodList.setLayoutManager(new LinearLayoutManager(this));

        // Atur RecyclerView untuk daftar minuman
        drinkList.setAdapter(kategoriAdapter);
        drinkList.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: app/src/main/java/com/dicoding/projectcapstone/kategori/KategoriActivity.kt
package com.dicoding.projectcapstone.kategori

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.dicoding.projectcapstone.R

class KategoriActivity : AppCompatActivity() {

    private lateinit var foodList: RecyclerView
    private lateinit var drinkList: RecyclerView
    private lateinit var kategoriAdapter: KategoriAdapter
    private val kategoriList = ArrayList<Kategori>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_kategori)

        foodList = findViewById(R.id.food_list)
        drinkList = findViewById(R.id.drink_list)

        kategoriList.add(
            Kategori(
                imageResource = R.drawable.ic_food,
                name = "Food",
                rating = 4.5,
                price = "$10",
                isOpen = true
            )
        )
        kategoriList.add(
            Kategori(
                imageResource = R.drawable.ic_drink,
                name = "Drink",
                rating = 4.0,
                price = "$5",
                isOpen = false
            )
        )

        kategoriAdapter = KategoriAdapter(kategoriList)

        foodList.adapter = kategoriAdapter
        foodList.layoutManager = LinearLayoutManager(this)

        // Atur RecyclerView untuk daftar minuman
        drinkList.adapter = kategoriAdapter
        drinkList.layoutManager = LinearLayoutManager(this)
    }
}
